code/game/machinery/lightswitch.dm

```diff
@@ -40,19 +40,19 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/light_switch, 26)
 	luminosity = (machine_stat & NOPOWER) ? 0 : 1
 
 /obj/machinery/light_switch/update_icon_state()
-	set_light(area.lightswitch ? 0 : light_on_range)
+	set_light(area?.lightswitch ? 0 : light_on_range)
```

Comment:
@optimumtact iirc you yelled at me two years ago for trying to add this exact check lol

I assume this is encountered during initialize / destroy?

Reply:
u little snitch lol

it seems very unlikely area would ever be null in normal situations, so I still don't agree with the nullability check

Reply:
I thought about it a bit, I think it'd be better to check for the initialized flag in gases like this 
since we don't want these cases running until we're out of initialize, we should just make that explicit

Reply:
god fucking damn it I can't. SHIT

Reply:
This sort of thing is really annoying because it's "this breaks if called before init is **mostly** done"
we need to be able to call it IN init, but we can't run it before area has been set, because if we do we get runtimes.
I can't even use the flag because the call I'm handing happens AFTER the flag is set

help

Reply:
Resolved, I didn't actually need to call update_appearance